### PR TITLE
10321 batch queries statement timeout

### PR DESCRIPTION
**Pull request by rafatower**
This change uses redis as storage directly, avoiding duplication and potential de-syncing with the metadata DB. Therefore this PR does not require a migration.

Check how it is used from the batch api:
https://github.com/CartoDB/CartoDB-SQL-API/blob/861b9bb037c1c17a9c0cdd1c2c98839bc893978e/batch/job_runner.js#L65-L79

I can write a few tests for this but prefer to get a first pass through your :eyes: before going any further

@jgoizueta and/or @ethervoid can you please review?

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1649,6 +1649,33 @@ def new_visualizations_version
     builder_enabled? ? 3 : 2
   end
 
+  def batch_queries_statement_timeout
+    if !defined?(@batch_queries_statement_timeout)
+      timeout_str = $users_metadata.HMGET(batch_limits_key, 'timeout').first
```
Comment (ethervoid) [inline app/models/user.rb]: Could this return nil and make the .first raise an exception?

File: app/models/user.rb [hunk 2]
```diff
@@ -1649,6 +1649,33 @@ def new_visualizations_version
...
+      @batch_queries_statement_timeout = timeout_str.nil? ? nil : timeout_str.to_i
+    end
+    @batch_queries_statement_timeout
+  end
+
+  def batch_queries_statement_timeout=(timeout_ms)
+    unless !timeout_ms.present? || timeout_ms.to_i > 0
```
Comment (ethervoid) [inline app/models/user.rb]: In case the timeout is nil, where is the default value? in sql-api?

Comment (ethervoid) [inline app/models/user.rb]: Ok, that's what I saw yesterday.

File: app/models/user.rb (hunk 1 above)
Comment (rafatower) [inline app/models/user.rb]:
```
irb(main):017:0> $users_metadata.HMGET('rails:users:cdb', 'geocoder_provider').first
=> "mapzen"

```

Comment (ethervoid) [inline app/models/user.rb]: Ok  cool!

**Comment by rafatower** [comment]
Changes since you've last seen this:
- Created tests. Most of them for real cases I had to test manually and tweak the code.
- Moved the code to a common module than `::User` and `Carto::User` can use.
- Please Hound

@jgoizueta or @ethervoid can you please take a final look? I'll wait for tests then deploy this. Thanks!

**Pull request merged**